### PR TITLE
init commit for issue 72 - API Gateway AuthType property - warning fo…

### PR DESCRIPTION
**Pull request by tmcelhattan**
Set a WARNING rule for AWS::ApiGateway::Method resource if `AuthorizationType` property is set to `NONE`. This property should be set to either of the following: `AWS_IAM`, `CUSTOM`, `COGNITO_USER_POOLS`.

### CODE REVIEW

File: lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W58'
+  end
+
+  def audit_impl(cfn_model)
+    violating_deployments = cfn_model.resources_by_type('AWS::ApiGateway::Method').select do |method|
+      method.authorizationType.nil? || method.authorizationType.to_s.downcase == 'none'
```
Comment (erickascic) [inline lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb]: technically nil.to_s == ''

Comment (erickascic) [reply lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb]: to_s.downcase is cool, but I believe we've got a bunch of prior art/style around using strcmpi

Comment (tmcelhattan) [reply lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb]: @erickascic  Just pushed up some slight changes. Is that what you had in mind?

Comment (erickascic) [reply lib/cfn-nag/custom_rules/ApiGatewayMethodAuthorizationTypeRule.rb]: ok i made a half stupid comment.  i like your original .nil? call better.  i was thinking nil being empty would fall through properly in the casecmp but it won't.  sorry for the runaround.

**Pull request merged**